### PR TITLE
Update api fix post tests

### PR DESCRIPTION
**Pull request by ndelvalle**
This PR adds:

* Format files using prettier (Wee need to update our eslint config to use the [recommended-configuration](https://prettier.io/docs/en/integrating-with-linters.html#recommended-configuration))
* Remove docs folder
* Create `create-request-body` functionality
* Update response object
* Use `nock` instead of `fetch-mock`
* Update / Improve `trae.post` tests

### CODE REVIEW

**Comment by gillchristian** [comment]
> We need to update our eslint config to use the recommended-configuration

Let's just remove the eslint check of prettier. We can automate prettier (run on commit), and run a prettier check on CI. But coupling it with eslint is quite annoying.

**Comment by ndelvalle** [comment]
@gillchristian I think we need an eslint config, prettier will not do everything eslint does. I think that using the recommended configuration will make things pretty easy. But I do not want to worry too much about it now.

**Comment by gillchristian** [comment]
I'm not talking about removing eslint. What I mean is to have them separated. Prettier for formatting, run check on CI to validate that everything is properly formatted. Eslint for code style conventions

Although between TS and prettier, there are just a few Eslint rules worth having.

**Comment by ndelvalle** [comment]
@gillchristian Even if we have them separated, we need to make sure our ESLINT rules like the way prettier format our code (That is not happening now), that is what I want with the recommended config.
I am not familiar with the TS + Prettier + ESlint configuration, I'll follow your lead on this.

File: lib/create-request-body.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+function isJSON(config: TraeSettings) {
+  const headers = config.headers;
+  // @ts-ignore
+  const key = Object.keys(headers).find(
+    // @ts-ignore
+    (header: any) => header.toLowerCase() === 'content-type',
+  );
+  // @ts-ignore
+  return key && headers[key].toLowerCase() === 'application/json';
+}
```
[the remark this answers is not in the document]
Comment (ndelvalle) [reply lib/create-request-body.ts]: @gillchristian I used `any` but the issue was not solved. I did not want to spend time debugging how to fix it. Let's fix it in another PR.

[the remark this answers is not in the document]
Comment (gillchristian) [reply lib/create-request-body.ts]: We should (which means I should) fix the types so we can start using it. Right now they are just blocking us 😖

**Pull request merged**